crates/provider-sdk/src/provider.rs

```diff
@@ -153,6 +153,20 @@ impl ProviderConnection {
         self.rpc_client.clone()
     }
 
+    /// Used for fetching the RPC client in order to make RPC calls
+    pub fn get_wrpc_client(&self, target: impl AsRef<str>) -> WrpcNatsClient {
```

Comment:
Once we get to working on tracing (as you mentioned) this function should  take trace context into account to set up headers

Reply:
Ah good point -- will make sure to add this

Reply:
made a note on the integeration issue!